nix/modules/fedimintd.nix

```diff
@@ -0,0 +1,233 @@
+    options = {
+
+      enable = mkEnableOption (lib.mdDoc "fedimint");
+
+      extraEnvironment = mkOption {
+        type = types.attrsOf types.str;
+        description = lib.mdDoc "Extra Environment variables to pass to the fedimintd.";
+        default = { };
+        example = { RUST_BACKTRACE = "1"; };
+      };
+
+      package = mkOption {
+        type = types.nullOr types.package;
+        default = pkgs.fedimintd or null;
+        defaultText = lib.literalExpression "pkgs.fedimint (after available)";
+        description = lib.mdDoc ''
+          Package of the fedimintd to use.
+        '';
+      };
+
+      p2p = {
+        openFirewall = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Opens port in firewall for fedimintd's p2p port";
+        };
+        port = mkOption {
+          type = types.port;
+          default = 8173;
+          description = lib.mdDoc "Port to bind on for p2p connections from peers";
+        };
+        bind = mkOption {
+          type = types.str;
+          default = "[::0]";
+          description = lib.mdDoc "Address to bind on for p2p connections from peers";
+        };
+        address = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "fedimint://p2p.myfedimint.com";
+          description = lib.mdDoc "Public address for p2p connections from peers";
+        };
+      };
+      api = {
+        openFirewall = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Opens port in firewall for fedimintd's api port";
+        };
+        port = mkOption {
+          type = types.port;
+          default = 8174;
+          description = lib.mdDoc "Port to bind on for API connections relied by the reverse proxy/tls terminator.";
+        };
+        bind = mkOption {
+          type = types.str;
+          default = "[::0]";
+          description = lib.mdDoc "Address to bind on for API connections relied by the reverse proxy/tls terminator. Usually starting with `fedimint://`";
+        };
+        address = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "wss://api.myfedimint.com";
+          description = lib.mdDoc "Public URL of the API address of the reverse proxy/tls terminator. Usually starting with `wss://`.";
+        };
+      };
+      bitcoin = {
+        network = mkOption {
+          type = types.str;
+          default = "signet";
+          example = "bitcoin";
+          description = lib.mdDoc "Bitcoin network to participate in.";
+        };
+        rpc = {
+          address = mkOption {
+            type = types.str;
+            default = "http://[::1]:38332";
+            example = "signet";
+            description = lib.mdDoc "Bitcoin node (bitcoind/electrum/esplora) address to connect to";
+          };
+          kind = mkOption {
+            type = types.str;
```

Comment:
We could make this an enum later on for better documentation of the available options.